lib/metadataTypes/MetadataType.js

```diff
@@ -403,31 +403,54 @@ class MetadataType {
+     * @param {boolean} [handleOutside] if the API reponse is irregular this allows you to handle it outside of this generic method
      * @returns {Promise} Promise
      */
-    static async createSOAP(metadataEntry) {
+    static async createSOAP(metadataEntry, overrideType, handleOutside) {
         try {
-            const res = await new Promise((resolve, reject) => {
-                this.removeNotCreateableFields(metadataEntry);
-                this.client.SoapClient.create(
-                    this.definition.type.charAt(0).toUpperCase() + this.definition.type.slice(1),
-                    metadataEntry,
-                    null,
-                    (error, response) => (error ? reject(error) : resolve(response))
-                );
-            });
-            Util.logger.info(
-                `- created ${this.definition.type}: ${metadataEntry[this.definition.keyField]}`
+            this.removeNotCreateableFields(metadataEntry);
+            let response;
+            await Util.retryOnError(
+                `Retrying to create ${this.definition.type}: ${
+                    metadataEntry[this.definition.nameField]
+                }`,
+                async () =>
+                    (response = await new Promise((resolve, reject) => {
+                        this.client.SoapClient.create(
+                            overrideType ||
+                                this.definition.type.charAt(0).toUpperCase() +
+                                    this.definition.type.slice(1),
+                            metadataEntry,
+                            null,
+                            (error, response) => (error ? reject(error) : resolve(response))
+                        );
+                    }))
             );
-            return res;
+            if (!handleOutside) {
+                Util.logger.info(
+                    `- created ${this.definition.type}: ${metadataEntry[this.definition.keyField]}`
+                );
+            }
+            return response;
         } catch (ex) {
-            Util.logger.error(
-                `- error creating ${this.definition.type}: ${
-                    metadataEntry[this.definition.keyField]
-                } (${ex.message})`
-            );
+            if (!handleOutside) {
+                let errorMsg;
+                if (ex.results && ex.results.length) {
+                    errorMsg = `${ex.results[0].StatusMessage} (Code ${ex.results[0].ErrorCode})`;
+                } else {
+                    errorMsg = ex.message;
+                }
+                Util.logger.error(
+                    `- error creating ${this.definition.type} '${
+                        metadataEntry[this.definition.keyField]
+                    }': ${errorMsg}`
+                );
+            } else {
+                throw ex;
+            }
 
-            return null;
+            return {};
```

Comment:
why change this?

Reply:
because the outside scripts try and test for returnValue.xyz and fail with an error if returnValue is null - instead of gracefully failing on not finding a value in xyz

Reply:
- [ ] return null when there is no result instead of empty object